Convex/clerk initial implement

kbeaverson opened the pull request:
- Updated landing page to sign in screen, put attendance behind log in authorization
- Implemented convex/clerk for nextjs following this walkthrough: https://docs.convex.dev/auth/clerk#nextjs
- Implement ConvexClientProvider
- Implement auth.config.ts
- Implement middleware.ts

Route protection reference: https://clerk.com/docs/references/nextjs/clerk-middleware

kbeaverson committed:
Initial implement

- Updated landing page to sign in screen, put attendance behind log in authorization
- Implemented convex/clerk for nextjs following this walkthrough: https://docs.convex.dev/auth/clerk#nextjs
- Implement ConvexClientProvider
- Implement auth.config.ts
- Implement middleware.ts

Route protection reference: https://clerk.com/docs/references/nextjs/clerk-middleware

ldfMion reviewed (changes requested):
I haven't had a chance to look super in depth but at a glance I agree with the following points brought up by Copilot:
- Use next/link instead of the anchor tag (), and make sure you use next/link and not the radixUI link component
- Remove use client from the root component and make the suggested client providers component (server rendering is very important in nextjs)

ldfMion commented:
I would also create a `dashboard/page.tsx` route to keep the account dashboard stuff and keep the signed/out logic there instead of the home page for now

kbeaverson committed:
Minor fixes

- Removed 'use client' fragment from earlier development stages
- Replace anchor usage with Link for page navigation/pre-rendering
- Remove unused imports from layout.tsx

kbeaverson committed:
Rework auth gating

- Restructured which components are blocked behind login status

ldfMion reviewed (changes requested):
- change navbar logic to a separate client component and make the app/page.tsx page a server component

ldfMion commented on web-app/app/page.tsx:
```diff
@@ -1,103 +1,76 @@
+    <>
+      <header className="w-full flex items-center justify-between py-4">
+        <h1 className="text-2xl font-semibold">ClubHub</h1>
+        <nav className="flex items-center gap-4">
```
Move this nav and its children to a a client component in a separate file. Use jt in the root layout.

kbeaverson replied:
Was "jt" supposed to be "it", or is there something I'm not yet familiar with called "jt"?

ldfMion commented on web-app/app/page.tsx:
```diff
@@ -1,103 +1,76 @@
+          </Authenticated>
+          <Unauthenticated>
+            <SignInButton mode="modal">
+              <button className="bg-foreground text-background px-3 py-1 rounded-md">
```
Use ShadCn button

ldfMion reviewed (changes requested):
Check comments

kbeaverson committed:
Refactor navbar

- Moved navbar from home page into separate component
- Installed shadcn button and used those for user auth buttons

kbeaverson committed:
Merge branch 'dev-db-auth' into kennyb/convex-clerk-auth

ldfMion reviewed (approved):
LGTM
